### DISCUSSION TITLE
.NET MAUI publish net8.0-ios, I create artifact .ipa,  i list it and i can see it , but it isn't saved.

### DISCUSSION THREAD

**Discussion by Siakis** [codemagic-ci-cd/codemagic-docs#2876] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

sign
build
publish

### Expected results

To correctly publish to testflight, I aslo ls $CM_BUILD_DIR/artifacts  and get -rw-r--r--   1 root     staff  48245469 Dec  9 08:13 Apografi.ipa

### Actual results

![Στιγμιότυπο οθόνης 2024-12-06 161423](https://github.com/user-attachments/assets/561788bd-66fd-4d7f-aaa6-b02855a51294)

### Build id (optional)

6752fe54c3636ba34975b3fa

**Comment by dam-ease** [comment]
Hello @Siakis
Looking through your builds, the ipa was generated but the artefact isn't available if you check your build details page.
You don't need to add the variable `CM_BUILD_DIR`. Also, is there a reason you are making use of the sudo command before the scripts, kindly share your use-case.
I'd suggest you follow the steps as in the [docs](https://docs.codemagic.io/yaml-quick-start/building-a-dotnet-maui-app/) to confirm this works at first try.

**Reply by Siakis** [reply]
Hey @dam-ease !
Why the artifact is not avaible? I have checked the build details page.
I used var CM_BUILD_DIR as I show on the sample codemagic dotnet project. 
I used sudo cause i though i needed superuser access.

**Reply by dam-ease** [reply, marked as answer]
Please can you take out the CM_BUILD_DIR from the var section of your codemagic.yaml. Also, remove sudo being used before the commands. Lastly, in your build section, kindly specify as:
```
-o ../artifacts
    artifacts:
        - ./artifacts/*.ipa
```
Here is the link to the sample dotnet project - https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/dotnet-maui/dotnet-maui-android-ios/codemagic.yaml

**Reply by Siakis** [reply]
Thank you! problem fixed! sudo was the problem, I guess it has smthn to do with path or permissions
